cmd/cloud-node-manager/app/config/config.go

```diff
@@ -32,6 +32,12 @@ type Config struct {
 	// the node name
 	NodeName string
 
+	// Specifies if node information is retrieved via IMDS or ARM.
+	UseInstanceMetadata bool
```

Comment:
there is already `"useInstanceMetadata": true`, shall we leverage that directly?
https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/examples/azure.json#L27

Reply:
I don't think we can, this flag is set at the time of cloud creation; we needed an additional flag to tell the cloud creation to use the azure.json config file or not. Thus I added the UseInstanceMetadata flag at the command level to guide the code to use the config file or not.